src/main/java/Duke.java

```diff
@@ -1,10 +1,294 @@
+import task.*;
+import storage.Storage;
+import ui.Ui;
+
+import java.util.*;
+
 public class Duke {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    public Duke() {
+
+        this.ui = new Ui();
+        this.storage = new Storage(System.getProperty("user.dir") + "/data/", "duke.txt");
+        this.taskList = storage.load();
+
+    }
+
+    public void run() {
+
+        Scanner sc = new Scanner(System.in);
+
+        ui.printWelcomeMessage();
+
+        boolean isExit = false;
+
+        while(sc.hasNextLine()) {
+
+            String input = sc.nextLine();
+
+            String[] tokens = input.split(" ", 2);
+
+            String command = tokens[0];
+            String argument = tokens.length == 2 ? tokens[1] : null;
+
+            if (command.equals("bye")) {
```

Comment:
Perhaps the code here can be replaced with a switch statement to improve readability. What do you think?

Reply:
Thank you for the idea! I will try to implement it